app/javascript/src/ChangePassword/components/ChangePassword.jsx

```diff
@@ -18,25 +18,111 @@ type Props = {
   errors: (?FlashMessage)[]
 }
 
+const PASSWORD = 'user[password]'
+const PASSWORD_CONFIRMATION = 'user[password_confirmation]'
+
 const validationConstraints = {
-  'user[password]': {
+  [PASSWORD]: {
     presence: true,
     length: { minimum: 1 }
   },
-  'user[password_confirmation]': {
+  [PASSWORD_CONFIRMATION]: {
     presence: true,
-    length: { minimum: 1 }
+    length: { minimum: 1 },
+    equality: PASSWORD
   }
 }
 
-const ChangePassword = ({ lostPasswordToken, url, errors }: Props) => {
-  const [isFormDisabled, setIsFormDisabled] = useState(true)
+const isFormValid = (formNode) => validateForm(formNode, validationConstraints)
+
+const useFormState = (formNode) => {
+  const [isDisabled, setIsDisabled] = useState(true)
+
+  const onChange = () => {
+    const errors = isFormValid(formNode)
+    setIsDisabled(!!errors)
+  }
+
+  return {
+    isDisabled,
+    onChange
+  }
+}
+
+const usePasswordState = () => {
+  const [value, setValue] = useState('')
+  const [isValid, setIsValid] = useState(undefined)
+
+  const onChange = (value, event) => {
+    setValue(value)
+    setIsValid(!!validateSingleField(event))
+  }
 
-  const onChange = (event) => {
-    const errors = validateForm(event.currentTarget, validationConstraints)
-    setIsFormDisabled(!!errors)
+  const onBlur = (event) => {
+    const errors = validateSingleField(event)
+    setIsValid(!!errors)
   }
 
+  return {
+    value,
+    isValid,
+    onChange,
+    onBlur
+  }
+}
+
+const getPasswordConfirmationErrors = (formNode) => {
+  const errors = isFormValid(formNode)
+  const passwordConfirmationError = (errors && errors[PASSWORD_CONFIRMATION]) && errors[PASSWORD_CONFIRMATION][0]
+  const passwordsMatchError = passwordConfirmationError && passwordConfirmationError.includes('is not equal to')
+  return {
+    hasError: !!passwordConfirmationError,
+    errorMessage: passwordsMatchError ? 'mustMatch' : 'isMandatory'
+  }
+}
+
+const compareInputsLength = (value) => {
+  const passwordInput = document.querySelector('input#user_password')
```

Comment:
Why don't we use a Node here instead of accessing the dom every time the input changes?